core/src/main/scala/org/apache/spark/scheduler/TaskSetBlacklist.scala

```diff
@@ -94,7 +96,9 @@ private[scheduler] class TaskSetBlacklist(val conf: SparkConf, val stageId: Int,
   private[scheduler] def updateBlacklistForFailedTask(
       host: String,
       exec: String,
-      index: Int): Unit = {
+      index: Int,
+      failureReason: Option[String] = None): Unit = {
```

Comment:
`failureReason` should always be present in this call, so it shouldn't be an `Option` as an arg to this method.

(I realize this is a bit of a pain as you have to modify all the call sites in tests, sorry about that).

Reply:
Actually , you are right.For feature completion i should modify this.